app.go

```diff
@@ -0,0 +1,170 @@
+	Status    string `json:"status"`
+	IsDeleted bool   `json:"isDeleted"`
+}
+
+var todos []Todo
+
+// GET /todos
+func getTodos(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(todos)
+}
+
+// GET /todos/{id}
+func getTodo(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	params := mux.Vars(r)
+	for _, item := range todos {
+		if id, err := strconv.Atoi(params["id"]); err == nil && item.ID == id {
+			json.NewEncoder(w).Encode(item)
+			return
+		}
+	}
+	json.NewEncoder(w).Encode(&Todo{})
+}
+
+func createTodo(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Contetn-Type", "application/json")
+	var todo Todo
+	_ = json.NewDecoder(r.Body).Decode(&todo)
+	todos = append(todos, todo)
+	json.NewEncoder(w).Encode(todo)
+}
+
+func softDeleteTodo(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Contetn-Type", "application/json")
+	params := mux.Vars(r)
+	var todo Todo
+	for idx, item := range todos {
+		if id, err := strconv.Atoi(params["id"]); err == nil && item.ID == id {
+			todos[idx].IsDeleted = true
+			todo = todos[idx]
+			// todos = append(todos[:idx], todos[idx+1:]...) // hard delete
+			break
+		}
+	}
+	json.NewEncoder(w).Encode(todo)
+}
+
+func markTodoDone(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Contetn-Type", "application/json")
+	params := mux.Vars(r)
+	var todo Todo
+	for idx, item := range todos {
+		if id, err := strconv.Atoi(params["id"]); err == nil && item.ID == id {
+			todos[idx].Status = "done"
+			todo = todos[idx]
+			break
+		}
+	}
+	json.NewEncoder(w).Encode(todo)
+}
+
+func markTodoPending(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Contetn-Type", "application/json")
+	params := mux.Vars(r)
+	var todo Todo
+	for idx, item := range todos {
+		if id, err := strconv.Atoi(params["id"]); err == nil && item.ID == id {
+			todos[idx].Status = "pending"
+			todo = todos[idx]
+			break
+		}
+	}
+	json.NewEncoder(w).Encode(todo)
+}
+
+func updateTask(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Contetn-Type", "application/json")
+	params := mux.Vars(r)
+	var todo Todo
+	for idx, item := range todos {
+		if id, err := strconv.Atoi(params["id"]); err == nil && item.ID == id {
+			todos[idx].Task = params["task"]
+			todo = todos[idx]
+			break
+		}
+	}
+	json.NewEncoder(w).Encode(todo)
+}
+
+func main() {
+	router := mux.NewRouter()
+
+	// TODO get data from DB or CSV
+	// todos = append(todos, Todo{ID: 10, Task: "Wash dishes", Status: "pending", IsDeleted: false})
+	// todos = append(todos, Todo{ID: 20, Task: "Make report", Status: "pending", IsDeleted: false})
+	// GET data from CSV
+	var filename string = "./data/todos.csv"
+	csvfile, err := os.Open(filename)
+	if err != nil {
+		log.Fatalln("Unable to open CSV file!", err)
+	}
+	fmt.Println("Loading records from ", filename)
+	r := csv.NewReader(csvfile)
+	var numOfRecords int = 0
```

Comment:
```suggestion
	numOfRecords := 0
```

In Go you can create variables using this shorthand.